TBTK/calc/TightBindingLib/src/FileWriter.cpp

```diff
@@ -1146,4 +1146,97 @@ bool FileWriter::exists(){
 	return exists;
 }
 
+void FileWriter::writeParameterSet(
+    const Util::ParameterSet *parameterSet,
+    std::string name,
+    std::string path
+){
+	init();
+
+	const int ATTRIBUTES_RANK = 1;
+	const hsize_t limits_dims[1] = {1};
+
+	try{
+		stringstream ss;
+		ss << path;
+		if(path.back() != '/')
+			ss << "/";
+		ss << name;
+
+		Exception::dontPrint();
+		H5File file(filename, H5F_ACC_RDWR);
+
+		DataSpace dataspace = DataSpace(ATTRIBUTES_RANK, limits_dims);
+		DataSet dataset = DataSet(file.createDataSet(name + "Int", PredType::STD_I64BE, dataspace));
+
+		for(int n = 0; n < parameterSet->getNumInt(); n++){
+			Attribute attribute = dataset.createAttribute(parameterSet->getIntName(n), PredType::STD_I64BE, dataspace);
+			int value = parameterSet->getIntValue(n);
+			attribute.write(PredType::NATIVE_INT, &value);
+		}
+
+		dataset = DataSet(file.createDataSet(name + "Double", PredType::IEEE_F64BE, dataspace));
+
+        for(int n = 0; n < parameterSet->getNumDouble(); n++){
+			Attribute attribute = dataset.createAttribute(parameterSet->getDoubleName(n), PredType::IEEE_F64BE, dataspace);
+			double value = parameterSet->getDoubleValue(n);
+			attribute.write(PredType::NATIVE_DOUBLE, &value);
+		}
+
+		hsize_t dimension = 2;
+        ArrayType dataTypeComplex(PredType::NATIVE_DOUBLE, 1, &dimension);
+		dataset = DataSet(file.createDataSet(name + "Complex", PredType::IEEE_F64BE, dataspace));
+
+        for(int n = 0; n < parameterSet->getNumComplex(); n++){
+			Attribute attribute = dataset.createAttribute(parameterSet->getComplexName(n), dataTypeComplex, dataspace);
+			complex<double> valueComplex = parameterSet->getComplexValue(n);
+			double value[2] = {real(valueComplex), imag(valueComplex)};
+			attribute.write(dataTypeComplex, value);
+		}
+
+		dataset = DataSet(file.createDataSet(name + "String", PredType::PredType::C_S1, dataspace));
+
+        for(int n = 0; n < parameterSet->getNumString(); n++){
+			string value = parameterSet->getStringValue(n);
+            StrType strDataType(PredType::C_S1, value.length());
```

Comment:
Make sure the naming convention agrees with that for complex in both FileReader and FileWriter.